server/manager/src/main/java/org/apache/accumulo/manager/compaction/coordinator/CompactionCoordinator.java

```diff
@@ -251,6 +250,18 @@ protected void startIdleCompactionWatcher() {
     ThreadPools.watchNonCriticalScheduledTask(future);
   }
 
+  protected void startGroupCheckThread() {
+    ScheduledFuture<?> future = schedExecutor.scheduleWithFixedDelay(() -> {
+      Set<String> compactorGroups = ExternalCompactionUtil.getCompactorAddrs(ctx).keySet();
```

Comment:
I looked at `ExternalCompactionUtil.getCompactorAddrs()` and I'm not clear on when the queue is removed from zookeeper based on this comment: 
https://github.com/apache/accumulo/blob/f3d5fb01d701a6932d37d2f67f52cc0eefa64d50/core/src/main/java/org/apache/accumulo/core/util/compaction/ExternalCompactionUtil.java#L121-L122

Looking at the code, if there are no live compactors for a given queue, the queue name is still added to the returned map with an empty list. 
https://github.com/apache/accumulo/blob/f3d5fb01d701a6932d37d2f67f52cc0eefa64d50/core/src/main/java/org/apache/accumulo/core/util/compaction/ExternalCompactionUtil.java#L117

If ZK does clean up the queue names quickly then this isn't an issue. but if not, the coordinator could also remove the group if the group's list of compactors is empty.

Reply:
> I'm not clear on when the queue is removed from zookeeper

ZooZap is the utility that will clean up ZooKeeper. It's typically used when the cluster is down for maintenance. Although I guess you could use it to clean up the compactors path if all of the Compactors are down.

Reply:
There is also [this](https://github.com/apache/accumulo/blob/2.1/server/compaction-coordinator/src/main/java/org/apache/accumulo/coordinator/CompactionCoordinator.java#L726) method that is called from a Thread [here](https://github.com/apache/accumulo/blob/2.1/server/compaction-coordinator/src/main/java/org/apache/accumulo/coordinator/CompactionCoordinator.java#L178) that runs every 5 minutes.